File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03290/p03290_num5_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int D, G;
    cin >> D >> G;

    vector<int> p(D), c(D);

    for (int i = 0; i < D; i++) {
        cin >> p[i] >> c[i];
    }

    int ans = 1e9;

    for (int i = 0; i < (1 << D); i++) {
        int score = 0;
        int problems = 0;
        int rest_max = -1;

        for (int j = 0; j < D; j++) {
            if ((i >> j) & 1) {
                score += 100 * (j + 1) * p[j] + c[j];
                problems += p[j];
            } else {
                rest_max = j;
            }
        }

        if (score < G) {
            int s1 = 100 * (rest_max + 1);

            if ((G - score + s1 - 1) / s1 < p[rest_max]) {
                problems += (G - score + s1 - 1) / s1;
            } else {
                continue;
            }
        }

        ans = min(ans, problems);
    }

    cout << ans << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03290/p03290_num5_parsed.py
# Solution

def solve(D, G, problems):
    min_problems = float('inf')
    for bitmask in range(1 << D):
        score = 0
        num_problems = 0
        max_bonus_idx = -1
        for i in range(D):
            if bitmask & (1 << i):
                score += problems[i][0] * 100 * (i + 1) + problems[i][1]
                num_problems += problems[i][0]
            else:
                max_bonus_idx = i
        
        if score < G:
            remaining = G - score
            bonus = problems[max_bonus_idx][0] * 100 * (max_bonus_idx + 1)
            if remaining % bonus == 0:
                num_problems += remaining // bonus
            else:
                num_problems += remaining // bonus + 1
        
        min_problems = min(min_problems, num_problems)
    
    return min_problems


# parsing input
D, G = map(int, input().split())
problems = []
for _ in range(D):
    p, c = map(int, input().split())
    problems.append((p, c))

# calling solve function
result = solve(D, G, problems)

# printing result
print(result)